web/src/engine/websites/AsuraScans.ts

```diff
@@ -47,8 +58,40 @@ export default class extends DecoratableMangaScraper {
     }
 
     public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
-        const chapters = await FetchCSS<HTMLAnchorElement>(new Request(new URL(manga.Identifier, this.URI)), 'div.scrollbar-thumb-themecolor a.block');
-        return chapters.map(chapter => new Chapter(this, manga, [manga.Identifier, 'chapter', chapter.pathname.match(/(\d+)+$/)[1]].join('/'), chapter.textContent.trim()));
+        const scripts = await FetchCSS<HTMLScriptElement>(new Request(new URL(`${manga.Identifier}`, this.URI)), 'script:not([src])');
+        const { chapters } = this.FindJSONObject<JSONChapters>(scripts, /is_early_access/, 'chapters');
+        return chapters.map(chapter => new Chapter(this, manga, [manga.Identifier, 'chapter', chapter.name.toString()].join('/'), `Chapter ${chapter.name}`));
     }
 
+    public override async FetchPages(chapter: Chapter): Promise<Page[]> {
```

Comment:
How about extracting image links with a delayed browser script instead?
Would be possible to avoid the fragile parsing of Next.js client-side rendering code.